kore/src/Kore/Step/SMT/Translate.hs

```diff
@@ -230,29 +228,39 @@ translatePredicateWith translateTerm predicate =
                 -- will fail to translate.
                 SMT.not <$> translateBool notChild
             ApplySymbolF app ->
-                (<|>)
-                    (translateApplication app)
-                    (translateUninterpreted SMT.tBool pat)
+                translateApplication (Just SMT.tBool) pat app
             DefinedF (Defined child) -> translateBool child
             _ -> empty
 
-    translateApplication :: Application Symbol p -> Translator m variable SExpr
+    translateApplication :: Maybe SExpr -> p -> Application Symbol p -> Translator m variable SExpr
     translateApplication
+        maybeSort
+        original
         Application
             { applicationSymbolOrAlias
             , applicationChildren
             }
-      = do
-        let translated = translateSymbol applicationSymbolOrAlias
-        sexpr <- maybe empty return translated
-        when (isNothing translated)
```

Comment:
From my understanding, if `translated` is `Nothing` this gets discarded before logging the warning. My changes should fix this.